src/Pimple/Container.php

```diff
@@ -233,8 +233,8 @@ public function extend($id, $callable)
             throw new \InvalidArgumentException(sprintf('Identifier "%s" does not contain an object definition.', $id));
         }
 
-        if (!is_object($callable) || !method_exists($callable, '__invoke')) {
-            throw new \InvalidArgumentException('Extension service definition is not a Closure or invokable object.');
+        if (!is_callable($callable)) {
```

Comment:
Here is why this is not done this way: we are storing parameters and services i in the same array. Using `is_callable` means that all of a sudden, a simple string store as a parameter can be interpreted as being a service (any built-in PHP function for instance). So, that's not possible. That's the reason why Pimple wants services to be defined as closures.